src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -170,4 +157,43 @@ public void InitializeDebugVarInfos(DebugVarInfo[] debugVarInfos)
             _debugVarInfos = debugVarInfos;
         }
     }
+
+    internal class WindowsMethodCodeNode : MethodCodeNode
+    {
+        public static readonly ObjectNodeSection StartSection = new ObjectNodeSection(".managedcode$A", SectionType.Executable);
+        public static readonly ObjectNodeSection ContentSection = new ObjectNodeSection(".managedcode$I", SectionType.Executable);
+        public static readonly ObjectNodeSection EndSection = new ObjectNodeSection(".managedcode$Z", SectionType.Executable);
+
+        public WindowsMethodCodeNode(MethodDesc method)
+            : base(method)
+        {
+        }
+
+        public override ObjectNodeSection Section
+        {
+            get
+            {
+                return ContentSection;
+            }
+        }
+    }
+
+    internal class UnixMethodCodeNode : MethodCodeNode
+    {
+        public static readonly ObjectNodeSection ContentSection = new ObjectNodeSection("__managedcode", SectionType.Executable);
```

Comment:
Is the name the only difference? We have been compensating for simple difference like this via if-checks, not by creating new types - e.g. look for `.IsWindows` under `src/ILCompiler.Compiler`.

target.IsWindows